cepheus-cep/src/main/java/com/orange/cepheus/cep/SubscriptionManager.java

```diff
@@ -241,8 +245,12 @@ private void periodicSubscriptionTask() {
      */
     private void subscribeProvider(Provider provider, SubscribeContext subscribeContext, Subscriptions subscriptions) {
         logger.debug("Subscribe to {} for {}", provider.getUrl(), subscribeContext.toString());
-
-        ngsiClient.subscribeContext(provider.getUrl(), null, subscribeContext).addCallback(subscribeContextResponse -> {
+      
+          if (StringUtils.isNotEmpty(provider.getServiceName())|| StringUtils.isNotEmpty(provider.getServicePath())) {
+                  httpHeaders = getHeadersForProvider(provider);
```

Comment:
If httpHeaders is per provider, then do not store them in the class... 
Later calls to subscribeProvider for another provider not defiing any serviceName/servicePath will reuse the httpHeaders from the last subscribed provider defining a serviceName/servicePath.

Reply:
I have defined httpheader in class so that if a subscribe request is done without serviceName and servicePath then value of httpheader remains null,so that it should  also perform the previous functionality.

Reply:
Hi @marcc-orange 

>>>If httpHeaders is per provider, then do not store them in the class...
Later calls to subscribeProvider for another provider not defiing any serviceName/servicePath will reuse the httpHeaders from the last subscribed provider defining a serviceName/servicePath.

I replied what i understood, please provide your opinion on this also correct my understanding if i am wrong.

Reply:
From your example, I think you understood what I mean by removing the httpHeaders.

It is simpler for me to show by code what I want:
```java
ngsiClient.subscribeContext(provider.getUrl(), getHeadersForProvider(), subscribeContext).addCallback(
...
```

and 

```java
ngsiClient.unsubscribeContext(provider.getUrl(), getHeadersForProvider(), provider.getSubscriptionId()).addCallback(
...
```
with getHeadersForProvider:

```java
/**
 * Returns the HTTP headers for a given provider, or null
 */
public HttpHeaders getHeadersForProvider(Provider provider) {
    String serviceName = provider.getServiceName();
    String servicePath = provider.getServicePath();
    
    // No headers when servicePath and serviceName are not defined
    if (serviceName == null || serviceName.isEmpty() || servicePath == null || servicePath.isEmpty()) {
        return null;
    }

    HttpHeaders httpHeaders = ngsiClient.getRequestHeaders(provider.getUrl());
    httpHeaders.add("Fiware-Service", serviceName);
    httpHeaders.add("Fiware-ServicePath", servicePath);
    return httpHeaders;
 }
```

Reply:
Thanks for prompt response.
I will do the same and commit the changes.